app/controllers/duties_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class DutiesController < ApplicationController
+  def index
+    @duties = Duty.all
+    tr = TimeRange.all.sort_by(&:start_time)
+    @first = tr.first.start_time
+    @last = tr.last.end_time
+    @Timeslot = Timeslot.all
+  end
+
+  def generate
```

Comment:
According to rubocop, Assignment Branch Condition size is too high -- need to refactor

Reply:
And, think about where you want to put this. Maybe in the model is better? :)